pkg/controller/job/backoff_utils.go

```diff
@@ -183,18 +183,35 @@ func getFinishedTime(p *v1.Pod) time.Time {
 }
 
 func getFinishTimeFromContainers(p *v1.Pod) *time.Time {
+	finishedTime := func(status v1.ContainerStatus) *time.Time {
+		if status.State.Terminated == nil ||
+			status.State.Terminated.FinishedAt.Time.IsZero() {
+			return nil
+		}
+		return &status.State.Terminated.FinishedAt.Time
+	}
+
 	var finishTime *time.Time
 	for _, containerState := range p.Status.ContainerStatuses {
-		if containerState.State.Terminated == nil {
+		fTime := finishedTime(containerState)
```

Comment:
Could we maybe commonize more the code for Containers and InitContainers? 
Maybe a function could either update the finish time in-place or return the new value for a slice of containers.

Reply:
> Maybe a function could either update the finish time in-place

Hard to do that because finishTime passed to the function will be nil at first, thus not able to be updated.
But I think we can just use the old logic, with a combined slice of both containers and init containers?

Reply:
> Hard to do that because finishTime passed to the function will be nil at first, thus not able to be updated.
I see, good point, how about a function which would take the previous finish time and return the new, say:
`latestFinishTime(prevFinishTime, containers) *time.Time`?

> But I think we can just use the old logic, with a combined slice of both containers and init containers?

Combining the slices would work for me too, but it is an extra copy of memory, so maybe can explore the alternative with the function approach?

Reply:
> latestFinishTime(prevFinishTime, containers) *time.Time

Yea, this could work.

Reply:
sgtm